File: testdata/fixtures/rust/rich.rs
#![allow(dead_code)]

#[derive(Debug, Clone)]
pub struct ThemePreview {
    value: String,
}

pub trait Renderable {
    fn render(&self) -> bool;
}

impl Renderable for ThemePreview {
    fn render(&self) -> bool {
        println!("{}", self.value);
        true
    }
}

macro_rules! themed {
    ($value:expr) => {
        ThemePreview { value: $value.into() }
    };
}

pub fn build() -> ThemePreview {
    let preview = themed!("Dracula");
    if preview.render() {
        preview
    } else {
        ThemePreview {
            value: String::from("fallback"),
        }
    }
}

File: testdata/fixtures/typescript/rich.ts
interface ThemePreview {
  title: string;
  enabled?: boolean;
}

export class Renderer {
  constructor(private readonly preview: ThemePreview) {}

  render(slug: string): string {
    const tags = ["dracula", slug].filter(Boolean);
    return `${this.preview.title}:${tags.join(",")}`;
  }
}
